src/spells.cpp

```diff
@@ -234,17 +234,12 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 	return count;
 }
 
-InstantSpell* Spells::getInstantSpellByIndex(const Player* player, uint32_t index)
+InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
-	uint32_t count = 0;
-	for (const auto& it : instants) {
-		InstantSpell* instantSpell = it.second;
-		if (instantSpell->canCast(player)) {
-			if (count == index) {
-				return instantSpell;
-			}
-			++count;
-		}
+	auto& it = instants.begin();
+	std::advance(it, spellId);
```

Comment:
Sure w/e you want :P 
But my plan with this PR was just to add basic support (also refering to mkalo's comment below) making it easier to add new stuff related to the spell class.
Ill do the changes tomorow.